src/Backend/ONE/ONEToolchain.ts

```diff
@@ -14,33 +14,121 @@
  * limitations under the License.
  */
 
+import {execSync} from 'child_process';
 import {Backend} from '../Backend';
 import {Command} from '../Command';
 import {Compiler} from '../Compiler';
 import {Executor} from '../Executor';
-import {Toolchains} from '../Toolchain';
+import {ToolchainInfo, Toolchains} from '../Toolchain';
+import {DebianArch, DebianRepo, DebianToolchain} from '../ToolchainImpl/DebianToolchain';
+import {Version} from '../Version';
 
 class ToolchainCompiler implements Compiler {
   private readonly toolchainTypes: string[];
+  private readonly toolchainName: string;
+  private readonly debianRepo: DebianRepo;
+  private readonly debianArch: DebianArch;
 
   constructor() {
-    this.toolchainTypes = [];
+    this.toolchainName = 'onecc-docker-test';
+    // we currently support only the latest version of one-compiler.
+    this.toolchainTypes = ['latest'];
+
+    // we temporary add personal PPA for test.
+    this.debianRepo =
+        new DebianRepo('http://ppa.launchpad.net/psalmist-kim/ppa/ubuntu', 'bionic', 'main');
+    this.debianArch = DebianArch.amd64;
   }
 
   getToolchainTypes(): string[] {
     return this.toolchainTypes;
   }
 
-  getToolchains(_toolchainType: string, _start: number, _count: number): Toolchains {
-    throw Error('Invalid getToolchains call');
+  getToolchains(_toolchainType: string, _start?: number, _count?: number): Toolchains {
+    if (_toolchainType !== 'latest') {
+      throw Error(`Invalid toolchain type : ${_toolchainType}`);
+    }
+
+    let isToolchain: Buffer;
+
+    try {
+      isToolchain = execSync(`apt-cache show ${this.toolchainName} > /dev/null 2>&1; echo $?`);
+    } catch (error) {
+      throw Error(`Getting ${this.toolchainName} package list is failed`);
+    }
+
+    if (parseInt(isToolchain.toString().trim()) !== 0) {
+      throw Error('The prerequisites are not met');
+    }
+
+    const availableToolchains = new Toolchains();
+
+    let isToolchainVersion: Buffer;
+    try {
+      isToolchainVersion = execSync(
```

Comment:
ditto. :)

Reply:
thank you! We're going to change them :)